File: Python3/800. Similar RGB Color.py
class Solution:
    def similarRGB(self, color: str) -> str:
        def find(colors: str) -> str:
            num = int(colors, 16)
            r = round(num / 17)
            return hex(r)[-1] * 2
            
        ans = '#'
        for i in range(1, 6, 2):
            ans += find(color[i: i+2])
        return ans

File: Java/800. Similar RGB Color.java
class Solution {
    public String similarRGB(String color) {
        StringBuilder sb = new StringBuilder();
        sb.append("#");
        
        for (int i=1; i<6; i+=2) {
            sb.append(find(color.substring(i, i+2)));
        }
        return sb.toString();
    }
    
    private String find(String color) {
        int num = Integer.parseInt(color, 16);
        int r = Math.round((float) num/17);
        return Integer.toHexString(r).repeat(2);
    }
    
}